testcode/tinygraphsolver.cpp

```diff
@@ -189,15 +192,45 @@ int main(int argc, char* argv[])
         truesol[5] = -2.83333333333333e+00;
     }
 
-    std::cout.precision(16);
-    sol.GetBlock(1).Print(std::cout, 1);
+    bool some_solver_fails = false;
+    for (auto& solver_pair : solver_to_name)
+    {
+        auto& solver = solver_pair.first;
+        auto& name = solver_pair.second;
+
+        std::cout << "\nSolving mixed graph Laplacian problem with "
+                  << name << " ..." << std::endl;
 
-    truesol -= sol.GetBlock(1);
-    double norm = truesol.Norml2();
-    std::cout << "Error norm: " << norm << std::endl;
+        // solve
+        sol = 0.0;
+        const_cast<MixedLaplacianSolver*>(solver)->SetPrintLevel(1);
```

Comment:
Avoiding this `const_cast` can be done by not having `const` in the map or setting the print level before placing into map.

 Otherwise, this looks good :+1: